src/libcollections/str.rs

```diff
@@ -1365,22 +1610,43 @@ impl str {
         core_str::StrExt::trim_right_matches(self, pat)
     }
 
-    /// Parses `self` into the specified type.
+    /// Parses this string slice into another type.
+    ///
+    /// Because `parse()` is so general, it can cause problems with type
+    /// inference. As such, `parse()` is one of the few times you'll see
+    /// the syntax affectionately known as the 'turbofish': `::<>`. This
+    /// helps the inference algorithm understand specifically which type
+    /// you're trying to parse into.
     ///
     /// # Failure
     ///
-    /// Will return `Err` if it's not possible to parse `self` into the type.
+    /// Will return `Err` if it's not possible to parse this string slice into
+    /// the desired type.
     ///
     /// # Example
     ///
+    /// Basic usage
+    ///
     /// ```
-    /// assert_eq!("4".parse::<u32>(), Ok(4));
+    /// let four: u32 = "4".parse().unwrap();
+    ///
+    /// assert_eq!(4, four);
     /// ```
     ///
-    /// Failing:
+    /// Using the 'turbofish' instead of annotationg `four`:
     ///
     /// ```
-    /// assert!("j".parse::<u32>().is_err());
+    /// let four = "4".parse::<u32>();
+    ///
+    /// assert_eq!(Ok(4), four);
+    /// ```
+    ///
+    /// Failing to parse:
+    ///
+    /// ```
+    /// let nope = "j".parse::<u32>();
+    ///
+    /// assert!(nope.is_err());
```

Comment:
Would be good to mention `FromStr` as a trait to look at when wanting to know what types and what kind of input for given types `parse` understands.